templates/rke2/overlay/opt/corral/rke2/init-cluster.sh

```diff
@@ -3,22 +3,38 @@
 CORRAL_api_host="${CORRAL_fqdn}"
 echo "corral_set api_host=${CORRAL_api_host}"
 
-mkdir -p /etc/rancher/rke2
-cat > /etc/rancher/rke2/config.yaml <<- EOF
-write-kubeconfig-mode: 644
+config="write-kubeconfig-mode: 644
```

Comment:
Does this change work for installing the "vanilla" Rancher package ?

I tried basically clone this into my branch made a couple lines changes (the airgap logic) but I end up with certificate errors.

The RKE2 cluster is technically up for me but I had to use insecure-skip-tls to connect to it.

I rolled back this to my original and it works without issues, I tried testing a bit more and I'm guessing something with the generated `config.yaml` ? I can't tell for sure.

But I wanted to double check with you to see if you're experiencing this too.

This is the initial error:
`Error: Kubernetes cluster unreachable: Get "https://<IP>:6443/version": tls: failed to verify certificate: x509: certificate is valid for 127.0.0.1, ::1, 192.168.13.174, 2600:1f1c:ab4:ee48:31ce:dfa8:4764:dbdb, 10.43.0.1, not <IP>
ERRO[0414] running /opt/corral/rancher/install-cert-manager.sh: Process exited with status 1`

And this is from the server node `journalctl -xe --no-pager --full`

`Apr 18 04:21:57 ip-192-168-18-46 rke2[2304]: time="2023-04-18T04:21:57Z" level=error msg="error syncing 'kube-system/rke2-metrics-server': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Apr 18 04:21:59 ip-192-168-18-46 rke2[2304]: time="2023-04-18T04:21:59Z" level=error msg="error syncing 'kube-system/rke2-ingress-nginx-controller-admission': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"`

Reply:
Hmmm ... weird, for me I was able to install a "vanilla" rancher with RKE2 and an air gap RKE2 rancher instance. But I had to use an updated cert and cert-key, since the one you gave me expired. Do you think that's it?

Reply:
Uhm, I think the certs are only for Rancher not for the RKE2 cluster. helm is not even able to get cert-manager installed in my case.
i'll have to keep digging

Reply:
Yeah I had the same issue when attempting to install the cert-manager, but when I updated the cert and the cert-key it worked.